packages/http/src/parameters.ts

```diff
@@ -116,13 +116,18 @@ function getOperationParametersForVerb(
         break;
       case "path":
         if (item.property.optional) {
-          diagnostics.add(
-            createDiagnostic({
-              code: "optional-path-param",
-              format: { paramName: item.property.name },
-              target: item.property,
-            }),
-          );
+          const uriParam = parsedUriTemplate.parameters.find((x) => x.name === item.property.name);
+          const isAutoRoute = operation.decorators.some((x) => x.definition?.name === "@autoRoute");
+
+          if (uriParam?.operator !== "/" && !isAutoRoute) {
+            diagnostics.add(
```

Comment:
We need to check as well with autoroute well produce the right uti template then